lib/main/MAVLink/mavlink_conversions.h

```diff
@@ -45,15 +45,15 @@ MAVLINK_HELPER void mavlink_quaternion_to_dcm(const float quaternion[4], float d
     double bSq = b * b;
     double cSq = c * c;
     double dSq = d * d;
-    dcm[0][0] = aSq + bSq - cSq - dSq;
-    dcm[0][1] = 2 * (b * c - a * d);
-    dcm[0][2] = 2 * (a * c + b * d);
-    dcm[1][0] = 2 * (b * c + a * d);
-    dcm[1][1] = aSq - bSq + cSq - dSq;
-    dcm[1][2] = 2 * (c * d - a * b);
-    dcm[2][0] = 2 * (b * d - a * c);
-    dcm[2][1] = 2 * (a * b + c * d);
-    dcm[2][2] = aSq - bSq - cSq + dSq;
+    dcm[0][0] = (float)(aSq + bSq - cSq - dSq);
```

Comment:
IMO it would be better to do calculations in `float`. Using `double` will pull in large math library and generate considerably slower code.
So the warning is good here